src/backend/metal/src/conversions.rs

```diff
@@ -155,6 +155,114 @@ impl PrivateCapabilities {
             }
         }
     }
+
+    pub fn map_format_properties(&self, format: MTLPixelFormat) -> Properties {
+        use hal::format::ImageFeature as feat;
+        use metal::MTLPixelFormat::*;
+        // use ImageFeature::SAMPLED_LINEAR as filter;
+        // use ImageFeature::STORAGE as write;
+        // use ImageFeature::COLOR_ATTACHMENT as color;
+        // use ImageFeature::COLOR_ATTACHMENT_BLEND as blend;
+
+        macro_rules! props {
```

Comment:
at the end of the day, we'll have more variation between formats than just a few bits of `optimal_tiling`, which this macro implies. Thus, the macro would have to grow and eventually become useless. Maybe we should start on the proper path right away, and do something like:
```rust
let defaults = Properties {
                    linear_tiling: If::empty(),
                    optimal_tiling: If::SAMPLED | If::BLIT_SRC | If::BLIT_DST,
                    buffer_features: BufferFeature::all(),
};
match format {
  some_format => Properties {
    optimal_tiling: If::SAMPLED_LINEAR | defaults.optimal_tiling,
    .. defaults
  },
}
```